docs: add site-admin capabilities (#8285)
sourcegraph/sourcegraph: doc/admin/index.md

# Site administration documentation

> Site administrators are the admins responsible for deploying, managing, and configuring Sourcegraph for regular users. Site administrator also come with [extra capabilities](capabilities.md) compare to regular users.

Reviewer: ```suggestion
Site administrators are the admins responsible for deploying, managing, and configuring Sourcegraph for regular users. They have [special privileges](privileges.md) on the Sourcegraph instance.
```
